lib/utils/device/deploy.ts

```diff
@@ -210,7 +206,10 @@ export async function deployToDevice(opts: DeviceDeployOptions): Promise<void> {
 	if (!opts.nolive) {
 		buildLogs = {};
 	}
-	const buildTasks = await performBuilds(
+
+	const { awaitInterruptibleTask } = await import('../helpers');
+	const buildTasks = await awaitInterruptibleTask<typeof performBuilds>(
```

Comment:
This adds CTRL-C support while performing a build on a local device (`balena push <ip-address>`), even before livepush starts watching the filesystem or showing device logs.